src/ipoplib.py

```diff
@@ -626,21 +747,46 @@ def parse_config():
     if not ("xmpp_username" in CONFIG and "xmpp_host" in CONFIG):
         raise ValueError("At least 'xmpp_username' and 'xmpp_host' must be "
                          "specified in config file or string")
-
+    # Use custom keyring incase of non-interactive environment
+    # (uses a fixed password)
+    if isinstance(keyring.get_keyring(),keyring.backends.file.EncryptedKeyring):
```

Comment:
Out of curiousity, can you expliain in which cases incur to use “keyring.backends.file.EncryptedKeyring”?